executor/runtime/docker/docker.go

```diff
@@ -2218,6 +2218,29 @@ func getExtraContainerEntrypoint(c *runtimeTypes.ExtraContainer) []string {
 	return c.ImageInspect.ContainerConfig.Entrypoint
 }
 
+func computeExtraContainersDockerCommand(c *runtimeTypes.ExtraContainer) []string {
+	if len(c.V1Container.Args) > 0 {
+		// If explicitly set k8s Args (docker CMD) on the pod, we need to return that, no question.
+		// This will be used as the CMD for the ENTRYPOINT already configured (which come from either the pod or the image)
+		return c.V1Container.Args
+	}
+
+	if len(c.V1Container.Command) > 0 {
+		// If there was a k8s Command (docker ENTRYPOINT) specified on the pod, then it would be inappropriate
+		// to use the k8s args (docker CMD) that came with the container, because they would have been originally
+		// written for a different docker ENTRYPOINT!
+		// This mimics the k8s and docker behavior documented here: https://docs.docker.com/engine/reference/run/#entrypoint-default-command-to-execute-at-runtime
```

Comment:
Thanks for fixing this comment - it is much clearer and clarifies that we are bringing the behavior into alignment with the stock kubelet and `docker run`, not deviating from their behavior.